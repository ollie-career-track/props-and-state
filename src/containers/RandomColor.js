import React, { Component } from 'react';
import ColorDisplay from '../components/colors/ColorDisplay';

export default class RandomColor extends Component {
  state = {
    backgroundColor: '',
    second: new Date()
  };

  componentDidMount() {
    this.timerID = setInterval(
      () => this.addSecond(),
      1000
    );
  }

  componentWillUnmount() {
    clearInterval(this.timerID);
  }

  changeColor() {
    const colors = ['red', 'blue', 'green', 'yellow'];
    const index = Math.floor(Math.random() * colors.length);
    let color = colors[index];

    return color;
  }

  addSecond() {
    this.setState({
      second: new Date(),
      backgroundColor: this.changeColor()
    });
  }

  render() {
    return (
      <>
        <ColorDisplay backgroundColor={this.state.backgroundColor} />
      </>
    );
  }
}
