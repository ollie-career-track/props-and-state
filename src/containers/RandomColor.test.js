import React from 'react';
import { shallow } from 'enzyme';
import RandomColor from './RandomColor';

describe('RandomColor component', () => {
  it('renders random color', () => {
    const wrapper = shallow(<RandomColor />);
    expect(wrapper).toMatchSnapshot();
  });
});
