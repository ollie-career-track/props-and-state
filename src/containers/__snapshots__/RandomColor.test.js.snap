// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RandomColor component renders random color 1`] = `
<Fragment>
  <ColorDisplay
    backgroundColor=""
  />
</Fragment>
`;
